### ISSUE TITLE
Picklable objects (currently throws an error)

### ISSUE THREAD

**Issue by AndreFCruz** [fairlearn/fairlearn#569]
#### Is your feature request related to a problem? Please describe.
```python
from sklearn.tree import DecisionTreeClassifier
from fairlearn.reductions import TruePositiveRateDifference, ExponentiatedGradient
clf = ExponentiatedGradient(DecisionTreeClassifier(), TruePositiveRateDifference())
clf.fit(X, y, sensitive_features=A)

import joblib
joblib.dump(clf, 'file.pkl')

# or
# import pickle; pickle.dumps(clf)
```
Throws the error: ```AttributeError: Can't pickle local object '_Lagrangian.best_h.<locals>.h'```

#### Describe the solution you'd like
I'd like to be able to pickle a fitted classifier/reductions object, or persist it in some other way.
I have searched the API docs and this doesn't seem to be a feature.

#### Additional context
It seems that the problem is some object that was defined locally and thus cannot be imported.
Maybe exposing this object in the package's namespace would be a solution (?)

**Comment by riedgar-ms** [comment]
I have a repro written as a test case, and [parameterised with both learners and moments](https://github.com/riedgar-ms/fairlearn/blob/riedgar-ms/pickle-expgrad-test/test/unit/reductions/exponentiated_gradient/test_exponentiatedgradient_pickle.py). I agree that this appears to be related to what the `_Lagrangian` is doing, although I've not had time to dig into that yet.

**Comment by adrinjalali** [comment]
making the objects pickle-able is really nice, but in the meantime you could try with `cloudpickle` and see if that helps for your usecase.

**Comment by AndreFCruz** [comment]
`cloudpickle` does seem to work, thanks!

**Comment by eddiebergman** [comment]
Just for future reference, the culprit function is here, it could be moved to global scope and it should pickle alright!

https://github.com/fairlearn/fairlearn/blob/594f42ba4e30c40ef1a61e739686c2d401a03bfa/fairlearn/reductions/_exponentiated_gradient/_lagrangian.py#L236-L242

**Comment by MiroDudik** [comment]
This is now addressed by #1167. Closing.

**Comment by pplonski** [comment]
Please release a new version with this fix.
